reactome_analysis_api/reactome_analysis_api/searcher/overview_fetcher.py

```diff
@@ -0,0 +1,68 @@
+import grein_loader
+import requests
+import json
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Fetcher():
+
+    @staticmethod
+    def get_available_datasets_grein(no_datasets: int = None) -> list:
```

Comment:
In terms of usage, I'd argue that the supported resources (in this case GREIN and ExpressionAtlas) should be hidden to the outside. Otherwise, we'd have to adapt quite a bit of code if we add a new resource in the future. Maybe add only one "public" function `get_available_datasets` that then internally fetches the datasets from the supported resources.

What do you think?